app/Core/Policies/DirectMessagePolicy.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Core\Policies;
+
+use App\Core\Models\User;
+use App\Core\Models\DirectMessage;
+use App\Authorization\Authorization;
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class DirectMessagePolicy
+{
+    use HandlesAuthorization;
+
+    /**
+     * Determine whether the user can delete the direct message.
+     *
+     * @param  DirectMessage $directMessage
+     * @param  User          $user
+     * @return bool
+     */
+    public function delete(User $user, DirectMessage $directMessage)
+    {
+        // TODO: I am not sure if it's correct
```

Comment:
just checking the user is the owner of the message is enough, so checking message sender_id against user is should be enough. `$user->id === $directMessage->sender_id`